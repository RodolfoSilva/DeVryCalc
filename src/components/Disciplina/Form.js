import React, { Fragment } from 'react';
import { TextField } from 'react-native-material-textfield';

const Nota = props => <TextField {...props} keyboardType="numeric" />;

const DisciplinaForm = ({ values, errors, onChange }) => (
  <Fragment>
    <TextField
      label="Nome da disciplina"
      value={values.name}
      onChangeText={value => onChange({ name: 'name', value })}
      error={errors.name}
    />
    <Nota
      label="Nota da AP1"
      value={values.ap1}
      onChangeText={value => onChange({ name: 'ap1', value })}
      error={errors.ap1}
    />
    <Nota
      label="Nota da AP2"
      value={values.ap2}
      onChangeText={value => onChange({ name: 'ap2', value })}
      error={errors.ap2}
    />
    <Nota
      label="Nota da AP3"
      value={values.ap3}
      onChangeText={value => onChange({ name: 'ap3', value })}
      error={errors.ap3}
    />
  </Fragment>
);

export default DisciplinaForm;
