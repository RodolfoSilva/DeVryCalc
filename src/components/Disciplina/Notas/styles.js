import { colors } from '../../../styles'

export const styleByNota = (nota = null) => {
  if (nota === null) return {}
  if (nota >= 5) return { color: colors.secondary }
  return { color: colors.danger }
}
