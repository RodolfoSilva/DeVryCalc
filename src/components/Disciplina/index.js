import React from 'react';
import { View, Text } from 'react-native';

import { Toolbar, Button, COLOR, Card } from 'react-native-material-ui';
const VerticalLine = () => (
  <View
    style={{
      width: 1,
      height: 20,
      backgroundColor: '#E4E4E5'
    }}
  />
);

const Nota = ({ title, value }) => (
  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
    <Text style={{ fontSize: 20 }}>{value === '' ? '_._' : value}</Text>
    <Text style={{ color: '#BBB', fontSize: 12 }}>{title}</Text>
  </View>
);

const RenderResult = (nota, legend) => {
  if (nota === null || nota > 10) {
    return null;
  }

  return (
    <React.Fragment>
      <Text style={{ fontSize: 30 }}>{nota}</Text>
      <Text style={{ fontSize: 11, textAlign: 'center' }}>{legend}</Text>
    </React.Fragment>
  );
};

const Result = ({ mediaFinal, notaParaAP1, notaParaAP2, notaParaAP3 }) => {
  const message = [
    RenderResult(mediaFinal, 'Média final'),
    RenderResult(notaParaAP1, 'Necessário tirar na AP1'),
    RenderResult(notaParaAP2, 'Necessário tirar na AP2'),
    RenderResult(notaParaAP3, 'Necessário tirar na AP3')
  ].find(result => result !== null);
  if (message) return message;

  return (
      <Text style={{ fontSize: 30 }}>_._</Text>
  );
};

const Disciplina = ({
  title,
  ap1,
  ap2,
  ap3,
  mediaFinal,
  notaParaAP1,
  notaParaAP2,
  notaParaAP3
}) => (
  <View style={{ padding: 8 }}>
    <View>
      <Text style={{ textAlign: 'left', fontWeight: '500' }}>{title}</Text>
    </View>
    <View style={{ alignItems: 'center', flexDirection: 'row' }}>
      <View
        style={{
          flex: 1,
          flexDirection: 'row',
          alignItems: 'center',
          minHeight: 90
        }}
      >
        <Nota title="AP1" value={ap1} />
        <VerticalLine />
        <Nota title="AP2" value={ap2} />
        <VerticalLine />
        <Nota title="AP3" value={ap3} />
      </View>
      <View
        style={{
          flex: 0,
          justifyContent: 'center',
          alignItems: 'center',
          width: 80
        }}
      >
        <Result {...{ mediaFinal, notaParaAP1, notaParaAP2, notaParaAP3 }} />
      </View>
    </View>
  </View>
);

export default Disciplina;
