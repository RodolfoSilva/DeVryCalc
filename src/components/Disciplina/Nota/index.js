import React from 'react'
import { Text, View } from 'react-native'
import PropTypes from 'prop-types'
import styles from './styles'

const Nota = ({ value = null, title, style, valueStyle, titleStyle }) => (
  <View style={[styles.container, style]}>
    <Text style={[styles.value, valueStyle]}>{value === null ? '_._' : value}</Text>
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </View>
)

Nota.propTypes = {
  value: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  style: PropTypes.object,
  valueStyle: PropTypes.func,
  titleStyle: PropTypes.func
}

export default Nota
