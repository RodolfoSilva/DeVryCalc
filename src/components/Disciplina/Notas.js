import React from 'react'
import { View } from 'react-native'
import Nota from './Nota'
import { colors } from '../../styles'
import { styleByNota } from './styles'

const Notas = ({ ap1, ap2, ap3, style }) => (
  <View style={style}>
    <Nota value={ap1} valueStyle={styleByNota(ap1)} title="AP1" />
    <Nota value={ap2} valueStyle={styleByNota(ap2)} title="AP2" />
    <Nota value={ap3} valueStyle={styleByNota(ap3)} title="AP3" />
  </View>
)

export default Notas
