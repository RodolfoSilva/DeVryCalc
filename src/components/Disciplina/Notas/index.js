import React from 'react'
import { View } from 'react-native'
import Nota from '../Nota'
import PropTypes from 'prop-types'
import { styleByNota } from './styles'

const Notas = ({ ap1 = null, ap2 = null, ap3 = null, style }) => (
  <View style={style}>
    <Nota value={ap1} valueStyle={styleByNota(ap1)} title="AP1" />
    <Nota value={ap2} valueStyle={styleByNota(ap2)} title="AP2" />
    <Nota value={ap3} valueStyle={styleByNota(ap3)} title="AP3" />
  </View>
)

Notas.propTypes = {
  ap1: PropTypes.number,
  ap2: PropTypes.number,
  ap3: PropTypes.number,
  style: View.propTypes.style
}

export default Notas
