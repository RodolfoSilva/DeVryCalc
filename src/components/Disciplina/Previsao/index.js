import React from 'react'
import PropTypes from 'prop-types'
import Nota from '../Nota'
import styles from './styles'

const Previsao = ({ value, title }) => (
  <Nota
    value={value}
    title={title}
    style={styles.container}
    valueStyle={styles.value}
  />
)

Previsao.propTypes = {
  value: PropTypes.number,
  title: PropTypes.string.isRequired
}
export default Previsao
