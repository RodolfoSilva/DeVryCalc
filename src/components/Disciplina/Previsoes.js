import React from 'react'
import { Text, View } from 'react-native'
import Previsao from './Previsao'
import { NotasCalculator } from '../../utils'

const Previsoes = ({ notas }) => {
  const mediaFinal = NotasCalculator.mediaFinal(notas)
  const notaParaAP1 = NotasCalculator.notaParaAP1(notas)
  const notaParaAP2 = NotasCalculator.notaParaAP2(notas)
  const notaParaAP3 = NotasCalculator.notaParaAP3(notas)

  return (
    <View>
      {mediaFinal !== null && <Previsao value={mediaFinal} title="Média final" />}
      {notaParaAP1 !== null && notaParaAP1 <= 10 && <Previsao value={notaParaAP1} title="Necessário tirar na AP1" />}
      {notaParaAP2 !== null && notaParaAP2 <= 10 && <Previsao value={notaParaAP2} title="Necessário tirar na AP2" />}
      {notaParaAP3 !== null && notaParaAP3 <= 10 && <Previsao value={notaParaAP3} title="Necessário tirar na AP3" />}
      {notaParaAP3 > 10 && <View style={{ alignItems: 'center', paddingVertical: 26 }}>
        <Text style={{ fontSize: 14, fontWeight: '500', color: '#333' }}>Não foi desssa vez, sorte na próxima....</Text>
      </View>}
    </View>
  )
}


export default Previsoes
