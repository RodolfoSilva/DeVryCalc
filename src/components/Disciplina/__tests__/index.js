import React from 'react'
import Previsoes from '../Previsoes'
import Notas from '../Notas'
import { Text } from 'react-native'
import Disciplina from '../'
import { shallow } from 'enzyme'

describe('Disciplina', () => {
  it('Deve renderizar a disiciplina', () => {
    const wrapper = shallow(<Disciplina titulo="Cálculo" id="dumb-id" notas={{}} onPress={() => {}} />)
    expect(wrapper.find(Text).first().prop('children')).toEqual('Cálculo')
    expect(wrapper.find(Previsoes).length).toEqual(1)
    expect(wrapper.find(Notas).length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza as notas', () => {
    const wrapper = shallow(<Disciplina titulo="Cálculo" id="dumb-id" notas={{ ap1: 10, ap2: 5, ap3: 3 }} onPress={() => {}} />)
    expect(wrapper.find(Notas).first().prop('ap1')).toEqual(10)
    expect(wrapper.find(Notas).first().prop('ap2')).toEqual(5)
    expect(wrapper.find(Notas).first().prop('ap3')).toEqual(3)
    expect(wrapper).toMatchSnapshot()
  })
})
