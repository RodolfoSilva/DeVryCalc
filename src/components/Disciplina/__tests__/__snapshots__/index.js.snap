// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disciplina Deve renderizar a disiciplina 1`] = `
<Card
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Cálculo
    </Text>
  </View>
  <Previsoes
    notas={Object {}}
  />
  <Notas
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  />
</Card>
`;

exports[`Disciplina Renderiza as notas 1`] = `
<Card
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Cálculo
    </Text>
  </View>
  <Previsoes
    notas={
      Object {
        "ap1": 10,
        "ap2": 5,
        "ap3": 3,
      }
    }
  />
  <Notas
    ap1={10}
    ap2={5}
    ap3={3}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  />
</Card>
`;
